### PR TITLE
Website:  Amend /docs/developers/dapp-developer/native:

### PR DESCRIPTION
**Pull request by zeuslawyer**
1. Fix typo
2. Fix slice of common.Hash slice syntax.
3. Clarify how the hex string for the `Transfer` event is computed so the code matches with the description.

### CODE REVIEW

File: docs/developers/dapp-developer/native.md [hunk 1]
```diff
@@ -86,17 +86,24 @@ nonce, err := cl.NonceAt(context.Background(), addr, big.NewInt(14000000))
...
 
 ```go
 blockNum, err := cl.BlockNumber(context.Background())
 if err != nil {
     return err
 }
+
+transferEventSignature := []byte("Transfer(address,address,uint256)")
+transferEventHash := crypto.Keccak256Hash(transferEventSignature)
+transferEventHashHexString = transferEventHash.Hex() // "0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef"
```
Comment (holiman) [inline docs/developers/dapp-developer/native.md:99]: The way you changed it, it becomes a bit overly complicated. Why convert it to hex-string, just to convert it back further below?

```suggestion
// "0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef"
transferEventHash := crypto.Keccak256Hash([]byte("Transfer(address,address,uint256)")) 
```

File: docs/developers/dapp-developer/native.md [hunk 2]
```diff
@@ -86,17 +86,24 @@ nonce, err := cl.NonceAt(context.Background(), addr, big.NewInt(14000000))
...
+
 q := ethereum.FilterQuery{
     FromBlock: new(big.Int).Sub(blockNum, big.NewInt(10)),
     ToBlock: blockNum,
-    Topics: [][]common.Hash{common.HexToHash("0xddf252ad1be2c89b69c2b068fc378daa952ba7f163c4a11628f55a4df523b3ef")},
+    Topics: [][]common.Hash{
+		  []common.Hash{common.HexToHash(transferEventHashHexString)}
+		},
```
Comment (holiman) [inline docs/developers/dapp-developer/native.md:106]:
```suggestion
    Topics: [][]common.Hash{
		  []common.Hash{transferEventHash}
		},
```

Comment (zeuslawyer) [reply docs/developers/dapp-developer/native.md]: @holiman 1000% agree thank you.  I've also updated for the typecasting of the blocknum value from uint to into to big.

Realised the old code wasnt compiling for me.

Comment (zeuslawyer) [reply docs/developers/dapp-developer/native.md]: @holiman  any chance you've got time to take a quick look? Other PRs to come soon as the docs are a bit outdated and I'm working on some training material for those new to geth so this is my way of learning to teach :)

**Review by MariusVanDerWijden — APPROVED** [review]

**Pull request merged**